### PR TITLE
feat: implement p/demo/stack

### PR DESCRIPTION
**Pull request by harry-hov**
This PR implements `p/demo/stack`

### CODE REVIEW

File: examples/gno.land/p/demo/stack/stack.gno [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package stack
+
+type Stack struct {
+	top    *node
+	length int
+}
+
+type node struct {
+	value interface{}
```
Comment (zivkovicmilos) [inline examples/gno.land/p/demo/stack/stack.gno:9]: Just curious about this - do we support generics for `.gno`?

You could easily turn this into:
```go
type node[T any] struct {
    value T
    prev *node
}
```

This way, we don't have to do the clunky type conversion each time you pop on the caller side

File: examples/gno.land/p/demo/stack/stack.gno [hunk 2]
```diff
@@ -0,0 +1,43 @@
+package stack
+
+type Stack struct {
+	top    *node
+	length int
```
Comment (zivkovicmilos) [inline examples/gno.land/p/demo/stack/stack.gno:5]: Just to be pedantic on the types - why not make this a `unit`?
You can never have negative stack size

File: examples/gno.land/p/demo/stack/stack.gno [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+	node := s.top
+	s.top = node.prev
+	s.length -= 1
+	return node.value
+}
+
+func (s *Stack) Push(value interface{}) {
+	node := &node{value, s.top}
+	s.top = node
```
Comment (zivkovicmilos) [inline examples/gno.land/p/demo/stack/stack.gno:41]: Nitpick, but you can inline this:
`s.top = &node{value, s.top}`

**Review by zivkovicmilos — APPROVED** [review]
Looks good 💯 

I've left a few comments, otherwise we're good to go

File: examples/gno.land/p/demo/stack/stack.gno (hunk 3 above)
Comment (harry-hov) [reply examples/gno.land/p/demo/stack/stack.gno:41]: Here, I'm bit inclined to keep this style.

File: examples/gno.land/p/demo/stack/stack.gno [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+func New() *Stack {
+	return &Stack{nil, 0}
+}
+
+func (s *Stack) Len() int {
+	return s.length
+}
+
+func (s *Stack) Top() interface{} {
```
[the remark this answers is not in the document]
Comment (harry-hov) [reply examples/gno.land/p/demo/stack/stack.gno:21]: Both "Peek"/"Top" are commonly used for stack. Both names are valid, but if you ask me, top makes more sense.

File: examples/gno.land/p/demo/stack/stack.gno (hunk 1 above)
Comment (harry-hov) [reply examples/gno.land/p/demo/stack/stack.gno:9]: I don't think we support generics yet.

File: examples/gno.land/p/demo/stack/stack.gno (hunk 2 above)
Comment (harry-hov) [reply examples/gno.land/p/demo/stack/stack.gno:5]: Almost all of the `Len` function/method returns `int`. Just to keep it consistent with other implementations let's use 'int'.

Also, you reminded me of this issue https://github.com/golang/go/issues/27460

**Review by moul — APPROVED** [review]

**Pull request merged**